### ISSUE TITLE
RemotingClient does not support setting a listener

### ISSUE THREAD

**Issue by funky-eyes** [apache/incubator-seata#7058]
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

RemotingClient不支持设置listener，当消息发出后channel断开时，无法通过listener进行快速响应失败。
RemotingClient does not support setting a listener. When the channel is disconnected after the message is sent, it cannot quickly respond to the failure through the listener

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

**Comment by lyl2008dsg** [comment]
I would like to claim this task

**Comment by YongGoose** [comment]
@lyl2008dsg 

Any updates?
If you're too busy to handle this issue, I'll take it.

**Comment by YongGoose** [comment]
@funky-eyes 

I worked on this in my personal repository https://github.com/YongGoose/incubator-seata/commit/9ae0709ddab229a058bcc31885f84519033cb1e0.

This task adds a `channel event listener` feature to Seata's `RemotingClient`.
With this enhancement, the system can now immediately detect events such as `network disconnections` or `message delivery failures` through `callbacks`, allowing custom logic—like retries or alerts—to be triggered as needed.
In short, it improves the handling of network issues and connection problems in a distributed transaction environment, making the system more responsive and flexible.

What do you think?

**Comment by YongGoose** [comment]
@funky-eyes 

I've created a PR!
- https://github.com/apache/incubator-seata/pull/7337

Please take a look and share your feedback when you have time. 🙂
